### ISSUE TITLE
pre-commit malfunctioning in 1.2-maint branch

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#7490]
after introduction of `pre-commit` (and locally installing the respective hook), one now can not commit in 1.2-maint branch:

```
fatal: could not open object name list: .git-blame-ignore-revs

fatal: could not open object name list: .git-blame-ignore-revs

fatal: could not open object name list: .git-blame-ignore-revs

fatal: could not open object name list: .git-blame-ignore-revs

14:36:28.206: [borg] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false add --ignore-errors -A -f -- src/borg/archiver.py
14:36:28.218: [borg] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false commit -F /private/var/folders/61/qtqwzmxd7t1fb19zck5k3q0h0000gn/T/git-commit-msg-.txt --
No .pre-commit-config.yaml file was found
- To temporarily silence this, run `PRE_COMMIT_ALLOW_NO_CONFIG=1 git ...`
- To permanently silence this, install pre-commit with the --allow-missing-config option
- To uninstall pre-commit run `pre-commit uninstall`
```

**Comment by ThomasWaldmann** [comment]
@Michael-Girma ^^^ can you have a look?

**Comment by Michael-Girma** [comment]
@ThomasWaldmann sure, let me take a look

**Comment by ThomasWaldmann** [comment]
I marked #7477 and #7489 for backporting to 1.2-maint, there is a bounty for backporting stuff, if you like!

**Comment by ThomasWaldmann** [comment]
Likely fixed by #7547.
